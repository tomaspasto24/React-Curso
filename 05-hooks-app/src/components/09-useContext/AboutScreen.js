import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser( {} );
    };

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />

            <button
                className='btn btn-primary'
                onClick={ handleClick }
            >
                Logout
            </button>
        </div>
    )
}
