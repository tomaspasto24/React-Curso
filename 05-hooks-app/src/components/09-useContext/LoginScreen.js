import React, { useContext } from 'react';
import { UserContext } from './UserContext';


export const LoginScreen = () => {

    const { user, setUser } = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            
            <button
                className='btn btn-primary'
                onClick={ () => setUser({ id: 2404, name: 'Fernando'}) }
            >
                Login
            </button>
        </div>
    )
}
