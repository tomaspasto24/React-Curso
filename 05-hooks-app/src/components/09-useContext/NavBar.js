import React from 'react';
import { NavLink } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const NavBar = () => {
    return (
        <>
            <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
                <div className="container-fluid">
                    <NavLink to='./' className="navbar-brand">useContext</NavLink>
                    <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        {/* activeClassName establece una clase si el URL coincide con el path. */}
                        <NavLink to='./' className="nav-link" aria-current="page">Home</NavLink>
                        <NavLink to='./About' className="nav-link">About</NavLink>
                        <NavLink to='./Login' className="nav-link">Login</NavLink>
                    </div>
                    </div>
                </div>
            </nav> 
        </>
    )
}
