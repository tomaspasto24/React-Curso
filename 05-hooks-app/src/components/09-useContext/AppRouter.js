import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
} from 'react-router-dom';
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
          <div>

              <NavBar />
            
              <div className='container'>
                <Routes>

                        {/* exact es para que sea la direcci√≥n exacta. 
                        element se ingresa el componente que tiene que ser renderizado. */}

                        <Route exact path='/' element={ <HomeScreen /> } />
                        <Route exact path='/about' element={ <AboutScreen /> } />
                        <Route exact path='/login' element={ <LoginScreen /> } />

                        {/* el path='*', se usa como por defecto. */}
                        <Route path='*' element={ <HomeScreen /> } />

                </Routes>
              </div>
          
          </div>  
        </Router>
    )
}
