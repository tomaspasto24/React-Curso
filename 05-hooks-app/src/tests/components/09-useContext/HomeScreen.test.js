import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en <HomeScreen />.', () => {
    const user = {
        name: 'Tom√°s',
        email: 'tomaspasto24@gmail.com'
    }

    // Se usa mount en vez de shallow porque mount renderiza todo, shallow solo el primer componente.
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <HomeScreen /> 
        </UserContext.Provider>
        ); 

    test('Debe de mostrarse correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    });
});