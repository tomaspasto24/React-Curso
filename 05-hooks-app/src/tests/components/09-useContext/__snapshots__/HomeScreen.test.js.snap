// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen />. Debe de mostrarse correctamente. 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Tom√°s",
   "email": "tomaspasto24@gmail.com"
}
  </pre>
</div>
`;
